File: api/src/Models/Booking.cs
namespace api.src.Models {
    public class Booking {
        public long BookingId { get; set; }
        public long UserId { get; set; }
        public long RoomId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Status { get; set; } = default!;
        public Member User { get; set; } = null!;
        public Room Room { get; set; } = null!;
    }
}

File: ClientApp/src/schemas/booking.ts
import * as v from "valibot";

export type Booking = v.InferInput<typeof BookingSchema>;
export const BookingSchema = v.object({
  bookingId: v.number(),
  userId: v.number(),
  userName: v.string(),
  roomId: v.number(),
  roomName: v.string(),
  startDate: v.string(),
  endDate: v.string(),
  status: v.string(),
});

export type BookingIdArray = v.InferInput<typeof BookingIdArraySchema>;
export const BookingIdArraySchema = v.array(v.number());
